cmd/crane/cmd/rebase.go

```diff
@@ -29,48 +35,184 @@ func NewCmdRebase(options *[]crane.Option) *cobra.Command {
 	rebaseCmd := &cobra.Command{
 		Use:   "rebase",
 		Short: "Rebase an image onto a new base image",
-		Args:  cobra.NoArgs,
-		RunE: func(*cobra.Command, []string) error {
-			origImg, err := crane.Pull(orig, *options...)
+		Args:  cobra.MaximumNArgs(1),
+		RunE: func(_ *cobra.Command, args []string) error {
+			if orig == "" {
+				orig = args[0]
+			} else if len(args) != 0 || args[0] != "" {
+				return fmt.Errorf("cannot use --original with positional argument")
+			}
+
+			// Parse and fetch the original image or index.
+			origRef, err := name.ParseReference(orig)
+			if err != nil {
+				return fmt.Errorf("parsing tag %q: %v", orig, err)
+			}
+			origDesc, err := remote.Get(origRef)
```

Comment:
Using `remote.Get` here breaks flags because we're not passing down `*options...`. 

I would use `crane.Head` to make sure it's not an index and then `crane.Pull` it otherwise. We waste a request but it doesn't really matter if this is fast. If you want to go fast, maybe add `crane.Get`.

Reply:
Done. Now using `crane.Head` and `crane.Pull`, but parsing the reference in order to determine whether it was a digest (and to modify it to set a tag instead) still requires calls to `name.ParseReference`. 🤔

Reply:
--insecure and --platform support now stolen from #1104